FV3/atmos_cubed_sphere/model/fv_dynamics.F90

```diff
@@ -300,6 +303,18 @@ subroutine fv_dynamics(npx, npy, npz, nq_tot,  ng, bdt, consv_te, fill,
                                  te_2d     => CCPP_interstitial%te0_2d     )
 #endif
 
+    integer, save :: id_dyn_core = -1, id_tracer_adv = -1, id_remap = -1, id_other = -1
+
+    if (id_dyn_core < 0) &
+        id_dyn_core = mpp_clock_id('   3.1.1.1-dyn_core', flags = clock_flag_default, grain=CLOCK_MODULE)
```

Comment:
I'm confused as to what is happening here, I think it's just not knowing what 'save' does. otherwise it looks like you are setting these variables to -1, then checking if they are < 0

Reply:
The save attribute makes sure the values are retained between one function call (fv_dynamics) to the next. Note that variables declared in a module (module variables) do not need the save attribute.